### PR TITLE
Command \center now has an option to center its content vertically

### PR DESCRIPTION
**Pull request by jodros**
I found this command was lacking an option to do the centering vertically (like asked here #475).

### CODE REVIEW

**Comment by jodros** [comment]
I'm unsure whether I should write the following after the settings:temporarily block

```lua
if SU.boolean(options.vertical) and then
      SILE.call("eject")
end
```

File: classes/plain.lua [hunk 1]
```diff
@@ -310,10 +310,14 @@ function class:registerCommands ()
...
   end)
 
-  self:registerCommand("center", function (_, content)
+  self:registerCommand("center", function (options, content)
     if #SILE.typesetter.state.nodes ~= 0 then
       SU.warn("\\center environment started after other nodes in a paragraph, may not center as expected")
     end
+    if SU.boolean(options.vertical) then
+      SILE.call("hbox")
+      SILE.call("vfill")
```
Comment (Omikhleia) [inline classes/plain.lua:319]: A vfill won't push the content middle of the page, _unless_ if it is a final page automatically vfill'ed (such as the final page of your example) or explicitly `\eject`'ed. Then another vfill is currently automatically added at the end of the page, resulting in centering.

Personally:
- I don't think that horizontal centering and vertical centering should be the same command
- I don' think the vfill added on final pages or via eject is actually good :)

**Review by alerque — CHANGES_REQUESTED** [review]
I think @Omikhleia is right here, writing direction (linewise) centering and advance direction (framewise) are very different operations and don't belong in the same command. In fact any sort of advance direction centering (absolute to page?, relative to frame), fill available remaining space?) is so complicated and the interactions with line spacing, start-of-frame and end-of-frame queues, and so forth is so complicated that this should go in a dedicated package and even if all possible modes are not implemented at first, some thought should be given to the command names(s) so that the rest of the scenarios can be implemented without stepping all over existing documents.

I don think that some centering handlers are sufficiently basic enough and used as building blocks for other commands that this is something the core distribution should provide, so adding a package in this repo is probably warranted.

**Comment by Omikhleia** [comment]
This old PR now has conflicts, and last discussions pointed it would not occur as-is.
I am closing it, but please feel free to re-open a dedicated issue and/or PR if a different approach should be discussed.

**Pull request closed without merging**